ublox/src/ubx_packets/packets.rs

```diff
@@ -402,6 +402,141 @@ enum NavStatusFlags2 {
     Inactive = 3,
 }
 
+/// Information message conifg
+#[ubx_packet_recv_send]
+#[ubx(
+    class = 0x06,
+    id = 0x2,
+    fixed_payload_len = 10,
+    flags = "default_for_builder"
+)]
+struct CfgInf{
+    protocol_id: u8,
+    reserved: [u8; 3],
+    #[ubx(map_type = CfgInfMask)]
+    inf_msg_mask_0: u8,
+    #[ubx(map_type = CfgInfMask)]
+    inf_msg_mask_1: u8,
+    #[ubx(map_type = CfgInfMask)]
+    inf_msg_mask_2: u8,
+    #[ubx(map_type = CfgInfMask)]
+    inf_msg_mask_3: u8,
+    #[ubx(map_type = CfgInfMask)]
+    inf_msg_mask_4: u8,
+    #[ubx(map_type = CfgInfMask)]
+    inf_msg_mask_5: u8,
+}
+
+#[ubx_extend_bitflags]
+#[ubx(from, into_raw, rest_reserved)]
+bitflags! {
+    /// `CfgInfMask` parameters bitmask
+    #[derive(Default)]
+    pub struct CfgInfMask: u8 {
+        const ERROR = 0x1;
+        const WARNING = 0x2;
+        const NOTICE = 0x4;
+        const DEBUG = 0x08;
+        const TEST  = 0x10;
+    }
+}
+
+#[ubx_packet_recv]
+#[ubx(
+    class = 0x4,
+    id = 0x0,
+    max_payload_len = 1240,
+    flags = "default_for_builder"
+)]
+struct InfError{
+    #[ubx(map_type = &str, 
+        may_fail, 
+        is_valid = inf::is_valid,
+        from = inf::convert_to_str,
+        get_as_ref)]
+    message: [u8; 0]
+}
+
+#[ubx_packet_recv]
+#[ubx(
+    class = 0x4,
+    id = 0x2,
+    max_payload_len = 1240,
```

Comment:
One loose end here is that there is no max payload length for these INF messages. How to express this? The code generator wants either fixed or max payload length.

Reply:
This is certainly suboptimal. I think this is fine for now though, eventually I will change the code generator to handle this better (and I'll update this when I get around to that).